pinc/ProjectTransition.inc

```diff
@@ -1159,10 +1182,10 @@ new ProjectTransition(
     [
         'project_restriction' => null,
         'action_name' => _('Make Project Available for Post-Processing'),
-        'confirmation_question' => '[default]',
+        'confirmation_question' => $pp_unavail_to_avail_confirm_question,
         'detour' => null,
-        'settings_template' => '',
-        'collateral_actions' => null,
+        'settings_template' => "checkedoutby=''",   // Clear reserved PPer
```

Comment:
This is the only part of the PR that I'm not 100% certain about, mostly because I don't fully grok all of the times that this transition is used. For instance, does this prevent transitioning a project that has a reserved PPer from being made available to that PPer?

Reply:
When a project is moved to Unavailable for PP in the first place, any assigned PPer is removed, so the only way there would be an assigned PPer is if the field was edited. At present, we don't even define a transition for Unavailable directly into PP checked out.

The db-req squirrels can fake this (for instance, a hospital project that the originally assigned PPer would like to have back) by using Project Jump to force the transition into PP checked out.

If a reserved PPer is in the PP slot when a project is Unavailable for PP, and you want a formal way to assign it to them, then we probably need a new transition from PP Unavailable to PP checked out, but if the transition is from Unavailable to Available, there should not be a PP assigned.

Tangent (and apologies if I should be bringing this up elsewhere): There's another problematical situation where the PM edits the PPer field and expects the book to be checked out to the person they assign it to. When they do this, it remains available in the pool unless someone notifies db-req about it. What I'd like to see there is blocking everyone but squirrels from editing the PP field if a project is in the PP pool, but that might need some discussion. I think that there was some discussion about an edit of a field in the project page not being something that should cause a project transition.

Reply:
@srjfoo probably knows more about that - I've never transitioned a project from Unavailable to either Available or Checked Out (except testing this). Directly below the snippet you've quoted is a transition from Unavailable to Checked Out, which says `checkedoutby='<WHO>'`. From my test, that appears to mean it will be checked out to the person making the change (assuming they accept the "It's unclear whether this transition should even be possible. Are you sure?" question). 

So, I think your query is that you might want to move an unavailable project with a reserved PPer into Available, so it is available for them to check out. With or without this commit, the PPer would have to then check it out themselves (or perhaps a squirrel could force that - not sure if a PM could force the move from Available to Checked Out). 

At present, the first step in that leaves the project in a weird state, where it is in the PP pool, but has a PPer. With this commit it will have no PPer. I don't think that affects whether the PPer would be able to check it out themselves. So, I think the two differences this commit makes are that it won't leave a project in Available but with a PPer assigned (probably desirable?) and that if a squirrel is jumping a project to get it to be checked out by the reserve PPer, and they go via Available, rather than direct from Unavailable to Checked Out, they would have to re-set the PPer's name.

Reply:
I think I got it, thanks. Both of you know more about this than I do so I'm happy to approve it. All of the other changes in this look great to me.

Reply:
> @srjfoo probably knows more about that - I've never transitioned a project from Unavailable to either Available or Checked Out (except testing this). Directly below the snippet you've quoted is a transition from Unavailable to Checked Out, which says `checkedoutby='<WHO>'`. From my test, that appears to mean it will be checked out to the person making the change (assuming they accept the "It's unclear whether this transition should even be possible. Are you sure?" question).

Thanks, Nigel, I was forgetting about that little transition. But yes, it doesn't check the project out to whoever (if anyone) is in the PP field, but to the person pushing the button.

> So, I think your query is that you might want to move an unavailable project with a reserved PPer into Available, so it is available for them to check out. With or without this commit, the PPer would have to then check it out themselves (or perhaps a squirrel could force that - not sure if a PM could force the move from Available to Checked Out).
>
> At present, the first step in that leaves the project in a weird state, where it is in the PP pool, but has a PPer. With this commit it will have no PPer. I don't think that affects whether the PPer would be able to check it out themselves. So, I think the two differences this commit makes are that it won't leave a project in Available but with a PPer assigned (probably desirable?) and that if a squirrel is jumping a project to get it to be checked out by the reserve PPer, and they go via Available, rather than direct from Unavailable to Checked Out, they would have to re-set the PPer's name.

A squirrel can force it, yes, with `project_jump`. And an unrestricted project manager (or PF or squirrel) can check it out themselves and then edit the PP field. A project in the PP pool with a username in the PP slot is not reserved for that PPer -- someone else can check it out, so it's better not to even allow there to be a username in the PP field if it's available for PP.